src/app/static/src/app/store/downloadResults/actions.ts

```diff
@@ -106,48 +106,78 @@ export const actions: ActionTree<DownloadResultsState, RootState> & DownloadResu
     },
 };
 
-export const getSummaryReportStatus = async function (context: ActionContext<DownloadResultsState, RootState>) {
-    const {state, dispatch} = context;
+export const getSummaryReportStatus = async function (context: ActionContext<DownloadResultsState, RootState>): Promise<void> {
+    const {state, dispatch, rootState, commit} = context;
     const downloadId = state.summary.downloadId;
     const response = await api<DownloadResultsMutation, DownloadResultsMutation>(context)
         .withSuccess(DownloadResultsMutation.SummaryReportStatusUpdated)
         .withError(DownloadResultsMutation.SummaryError)
         .get<ModelStatusResponse>(`download/status/${downloadId}`)
-    await getADRUploadMetadata(response, dispatch);
+
+    if (response && response.data?.done) {
+        await getADRUploadMetadata(response, dispatch).then(() => {
+            commit({
+                type: DownloadResultsMutation.SummaryMetadataError,
+                payload: rootState.metadata.adrUploadMetadataError
+            });
+        });
+    }
 };
 
-export const getSpectrumOutputStatus = async function (context: ActionContext<DownloadResultsState, RootState>) {
-    const {state, dispatch} = context;
+export const getSpectrumOutputStatus = async function (context: ActionContext<DownloadResultsState, RootState>): Promise<void> {
+    const {state, dispatch, rootState, commit} = context;
     const downloadId = state.spectrum.downloadId;
     const response = await api<DownloadResultsMutation, DownloadResultsMutation>(context)
         .withSuccess(DownloadResultsMutation.SpectrumOutputStatusUpdated)
         .withError(DownloadResultsMutation.SpectrumError)
         .get<ModelStatusResponse>(`download/status/${downloadId}`);
-    await getADRUploadMetadata(response, dispatch);
+
+    if (response && response.data?.done) {
+        await getADRUploadMetadata(response, dispatch).then(() => {
+            commit({
```

Comment:
Committing this mutation really ought to be the `withError` clause of the `getADRUploadMetadata` action - is there any reason to separate them here? In fact, why do we even need to do a separate mutation here to add it to the download state if we're already going to have the error available in the metadata state? So that it's available in 'summary' and 'spectrum' in the component? I think it would be better to just to have the values in the state once rather than having to manage duplication here. You could add a computed into the component to arrange them in a nice way if youlike.

Reply:
Sorry, yes, didn't appreciate that you'd be overwriting the error for each type so you do need to commit into a different bit of the state. I still don't like this commit happening here, though - it's associated with getting ADR metadata so it feels like it should be part of that action to make any necessary commits. 

Maybe we should just save separate errors by upload type in the adrUploadMetadataError rather than a single one, if we need to keep them by type now, rather than also adding to the DownloadResults state? As far as I can tell we don't seem to do anything with that one in the metadata state anyway so it's not doing anything useful for us. getADRUploadMetadata action could take type (summary/spectrum) as parameter so would know which upload type to commit for,

Reply:
OK, I think you're probably right that this is the least bad way. But would you mind putting a brief comment here about the need for this commit as this is certainly not the standard way to do commits following actions